Add the prompt "Are you satisfied with WooPayments?" for eligible merchants, behind feature flag

Jinksi opened the pull request:
Resolves #10326
Partially resolves #10331 by adding tracks events for some interactions introduced in this PR

#### Changes proposed in this Pull Request

* Introduces a snackbar prompt that asks a merchant "Are you satisfied with WooPayments?".
* Action buttons currently do nothing except record tracks events. Action handlers will be implemented in:
	* #10327 
	* #10328
* Dismiss state is stored locally, to be persisted in user preferences in #10329 
* This prompt is only visible if the development feature flag `_wcpay_feature_prompt_merchant_for_review` is enabled – to be removed from client on launch
* Only renders for merchants eligible as per the WordPress.org 2025 review campaign eligibility flag from account metadata → `wcpaySettings.accountStatus.campaigns.wporgReview2025`
  * Default campaign status is inactive (false)
  * See https://github.com/Automattic/transact-platform-server/pull/7246 for implementation of this flag
* Only renders when no other `core/notices` snackbars are present.



![image](https://github.com/user-attachments/assets/33526e3c-c430-4c8f-a895-cff149630a24)


Pages where this is visible:
* Payments overview
* Capital loans list
* Payout list
* Payout detail
* Dispute list (not dispute challenge form, to not bother merchants while responding to dispute)
* Documents list
* Transactions list
* Transactions detail


#### Temporary snackbar implementation / hack
* The snackbar was heavily customised using custom React markup to workaround the following limitations:
  *	The bundled `@wordpress/components` version in WooPayments doesn't support explicit dismiss button (`x`)
  *	The `@wordpress/components` `Snackbar`/`SnackbarList` doesn't allow two actions to be rendered
  * The `core/notices` `createNotice()` only accepts a single action and no custom React markup
* This prompt is a "shadow" SnackbarList that uses a React portal to place this snackbar in the same position that WC-wide `createNotice()` snackbars are rendered.
* This workaround/hack will remain in place until, either:
	* We remove the prompt at the end of the campaign
	* The required changes to Snackbar are introduced to Gutenberg and WooPayments unbundles/upgrades to the required `wp.components` version

> [!IMPORTANT]
>  This prompt is only visible if the development feature flag `_wcpay_feature_prompt_merchant_for_review` is enabled

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* ~Checkout the transact-platform-server PR https://github.com/Automattic/transact-platform-server/pull/7246~ this has been deployed
* Enable the dev feature flag `_wcpay_feature_prompt_merchant_for_review`.
	* `update_option( '_wcpay_feature_prompt_merchant_for_review', '1' )`
* Ensure your store meets the campaign elibility criteria from the server PR linked above, or manually set `'wporgReview2025' => true` [here](https://github.com/Automattic/woocommerce-payments/blob/61fc121aec4d5718734f5a720517e694875c70de/includes/class-wc-payments-account.php#L378)
* Visit Payments → Overview or any of the pages mentioned in the PR description.
* Ensure you see the new snackbar prompt.
* Ensure the prompt disappears if other snackbar notices are presented, e.g. when exporting CSV reports from the Transactions List screen.
* Ensure the snackbar is dismissed when clicking yes, no, X or the snackbar itself.
* Ensure tracks events are recorded when the snackbar is rendered and for each interaction event.
* Visiting other Payments screens listed above will render the bar after dismissing: this dismiss behaviour will be persisted in a separate issue #10329 
* With the **dev feature flag** or **campaign eligibility flag** disabled, you should not see any notice on any WooPayments page, e.g. Overview, Transactions, Disputes, etc.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : **QA Testing Not Applicable** since eligibility is difficult to prepare, manual internal testing is suitable.
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable. **Not a critical flow**
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable. **What's new in 9.1.0 post: paJDYF-gXj-p2**

Jinksi committed:
Add support for WordPress.org Review 2025 campaign flag

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `10370` or branch name `add/10326-add-prompt-for-review` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/10326-add-prompt-for-review) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 704162784acc3e628649fb899bb0165ee6711280
- Build time: 2025-03-04 02:02:11 UTC

_Note: the build is updated when a new commit is pushed to this PR._

brucealdridge commented on includes/class-wc-payments-account.php:
```diff
@@ -372,6 +372,9 @@ public function get_account_status_data(): array {
 				'declineOnCVCFailure' => $account['fraud_mitigation_settings']['cvc_check_enabled'] ?? null,
 			],
+			'campaigns'             => [
+				'wporgReview2025' => $account['eligibility_wporg_review_campaign_2025'] ?? false,
```
This value is set to false by default so if the flag doesn't exist (or is removed from the server) the campaign will not be active.

Jinksi replied:
This is worthy of a code comment → added in 26ab129

Jinksi committed:
Add comment to clarify WordPress.org Review 2025 campaign flag behavior

Jinksi committed:
Add merchant feedback prompt feature flag

Jinksi committed:
Add merchant feedback prompt feature flag to wcpaySettings

Jinksi committed:
Add initial merchant feedback prompt component and integration

- Create merchant feedback prompt component
- Show merchant feedback prompt on overview page with feature flag

Jinksi committed:
Add feedback prompt with dismissal functionality, persisted in local component state

Jinksi committed:
Show prompt on many Payments screens

Jinksi committed:
Remove test notice

Jinksi committed:
Merge branch 'develop' into add/10326-add-prompt-for-review

Jinksi committed:
Refactor: Improve merchant feedback prompt portal documentation and simplify portal rendering

Jinksi committed:
Rename SnackbarPortal to WCFooterPortal for clarity

Jinksi committed:
Add comment explaining manual explicit dismiss button

haszari commented on client/globals.d.ts:
```diff
@@ -17,6 +17,7 @@ declare global {
 			paymentTimeline: boolean;
 			isDisputeIssuerEvidenceEnabled: boolean;
 			isPaymentOverviewWidgetEnabled?: boolean;
+			isMerchantFeedbackPromptEnabled: boolean;
```
Is this the same as the [server side eligibility flag](https://github.com/Automattic/transact-platform-server/pull/7246) (i.e. is this a placeholder)?

Jinksi replied:
This is a client-only dev flag, needed to allow us to push in-development code to trunk. I've renamed to clarify in in 6d0e870.

haszari commented on client/globals.d.ts:
```diff
@@ -69,6 +70,9 @@ declare global {
 			};
+			campaigns: {
+				wporgReview2025: boolean;
+			};
```
Do we need to document `campaigns` somewhere? Feels like a new concept. Also will all the children of it follow similar rules (e.g. inactive if MIA etc)?

haszari replied:
Should we use `campaigns.wporgReview2025` to track if merchant has responded or dismissed?

If `campaigns.wporgReview2025` stays boolean, how can we rename it so it's super obvious how it works / what it is?

Jinksi replied:
I've added some comments in ed1e318 to clarify this.

haszari replied:
Thanks – the new comments help give some clues to context.

haszari commented on includes/class-wc-payments-features.php:
```diff
@@ -393,4 +395,13 @@ private static function is_woopayments_gateway_enabled() {
 	}
+
+	/**
+	 * Checks if the merchant feedback prompt feature flag is enabled.
```
Might as well use this comment to clearly define/document this feature flag.

Is this a temporary feature flag, to allow us to ship the code and opt-in manually to test (e.g. on local or staging sites)? 

When the dust settles on this PR will be useful to document the various flags at plugin and service level, and which are for development purposes vs. to manage eligibility and dismiss state (etc) after launch.

Jinksi replied:
I've added a comment here to clarify that this is a temporary, development feature-flag.

Jinksi replied:
I've added a comment here and renamed this across the codebase to clarify that this is a temporary, development feature-flag in 6d0e870

haszari reviewed (commented):
Left some comments – mainly me exploring the different flags and getting a feel for this. 

The key feedback (likely already on your radar, WIP!) is to clearly document the various flags (across plugin and service) and which are for production / launch vs to help us during development.

Jinksi committed:
Record tracks events on interactions

Jinksi committed:
Dismiss prompt on element click, consistent with other snackbars

Jinksi committed:
Add integration testing of component

Jinksi committed:
Simplify by moving dev feature flag check to MerchantFeedbackPrompt component

Jinksi committed:
Rename merchant feedback prompt feature flag to clarify development purpose

Jinksi committed:
Add code comments explaining the campaign flag

Jinksi committed:
Add account campaign eligibility check to merchant feedback prompt

Jinksi committed:
Add placeholder changelog entry

Jinksi committed:
Show prompt on payment details screen

Jinksi committed:
Refactor merchant feedback prompt to use wrapper component

Jinksi committed:
Add comments explaining temporary snackbar implementation details

Jinksi committed:
Clarify the `dismissPrompt` callback function

ciampo commented on client/merchant-feedback-prompt/index.tsx:
```diff
@@ -0,0 +1,223 @@
+						 * See https://github.com/WordPress/gutenberg/blob/c300edfebb48f79f6f0f6643ce04dd73303c5fcb/packages/components/src/snackbar/index.tsx#L119-L126
+						 */
+						content: (
+							<Flex
```
Using `VStack` could be a simpler way to achieve the same layout

Jinksi replied:
Thanks @ciampo, TIL about `VStack`!

Unfortunately, our outdated bundled version of `@wordpress/components` (19.8.5) and the [deprecated `@types/wordpress__components`](https://www.npmjs.com/package/@types/wordpress__components/v/19.10.0) doesn't allow me to import `VStack` or `__experimentalVStack`, so I'd need to either override the type definitions manually or update the components or `@types` library versions, both out of scope for this UI change.

I'd also be concerned about using an early (3+ years old) version of an experimental component, when `Flex` seems to do the trick well enough.

ciampo replied:
Got it. I strongly recommend updating `@wordpress/*` dependencies, since, as you also mention, they are 3+ years old. The package has been shipping its own types for a while, on top of adding several improvements and features.

Jinksi replied:
Yes 100% agree @ciampo, these problems are only going to get worse as time goes on. This is a known issue that we're prioritising in these P2 posts:

[Aligning WooPayments with the WordPress component package](paJDYF-gne-p2)
and 
[Spike: How WooPayments imports, bundles, and overrides styles from WP components](paJDYF-gpC-p2)

jessy-p committed:
Merge branch 'develop' into add/10326-add-prompt-for-review

jessy-p reviewed (commented):
Tested ✅ 
<img width="463" alt="image" src="https://github.com/user-attachments/assets/827b4348-ccd3-40b9-abcb-68ebbfd77731" />

As already mentioned in the issue,  snackbar is re-rendered on refresh of page even after dismissing, which will be fixed in #10329

Jinksi committed:
Merge branch 'develop' into add/10326-add-prompt-for-review

Jinksi committed:
Add more details explaining the Snackbar and portal workarounds

Jinksi commented:
PS @Automattic/helix if anyone approves this, please feel free to merge to help unblock subsequent project PRs 🚀

Jinksi committed:
Merge branch 'develop' into add/10326-add-prompt-for-review

Jinksi committed:
Fix missing `>` after merge conflict resolution

haszari commented on client/globals.d.ts:
```diff
@@ -20,6 +20,10 @@ declare global {
 			isDisputeIssuerEvidenceEnabled: boolean;
 			isPaymentOverviewWidgetEnabled?: boolean;
+			/**
+			 * The development feature flag for the merchant feedback prompt.
```
Maybe link to a GitHub issue or something to define this in simple terms? We currently know what "merchant feedback prompt" means but others might not. The comment doesn't add much, says same info as the name 😁

Jinksi replied:
Resolved in 704162784acc3e628649fb899bb0165ee6711280

haszari commented on client/merchant-feedback-prompt/index.tsx:
```diff
@@ -0,0 +1,226 @@
+ * A react portal for the merchant feedback prompt.
+ * This is used to render the custom snackbar prompt in the WC footer component, consistent with where WC notices (snackbars) are rendered.
+ *
+ * HACK: This is a temporary solution until Gutenberg Snackbar component and `createNotice` accept two actions.
```
Is the "until" part misleading here? 

My understanding is …

- We need to implement this ourselves now (hack / copy / fork).
- We will remove the code before we could potentially use any upstream changes.

Jinksi replied:
> We will remove the code before we could potentially use any upstream changes.

I'm not ruling out a scenario where we decide to keep this prompt going, so added this note there in case this code remains when the time comes to update the upstream component library.

I'm also optimistic that we'll be addressing the unbundling of WooPayments before this campaign ends.

haszari commented on client/merchant-feedback-prompt/index.tsx:
```diff
@@ -0,0 +1,226 @@
+ *
+ * This temporary workaround will remain in place until either:
+ * - Gutenberg Snackbar component is updated to accept two actions and we can use `core/notices` `createNotice()` to render the snackbar, or
+ * - The campaign for this prompt is complete and we can remove this code entirely.
```
Ah this clears it up, but maybe we can just say it's temporary and we're gonna remove it. Good to mention the possibility of using core components in future, but that seems unlikely.

Jinksi replied:
As I mention above, I don't think it is unlikely that we'll unbundle and update the upstream component in the near future, and/or we keep this prompt in place past the campaign.

Since this is a simple code comment, there is no harm in mentioning this case to code readers. I'll try to tidy up these comments to reduce the burden.

Jinksi replied:
I've clarified and simplified these code comments in 8879ba9

haszari commented on client/merchant-feedback-prompt/index.tsx:
```diff
@@ -0,0 +1,226 @@
+	// Get the core notices, which we'll use to ensure we're not rendering the prompt if there are other notices being displayed.
+	const coreNotices = useSelect(
+		( select ) =>
+			select( 'core/notices' ).getNotices() as NoticeList.Notice[]
```
I think of this code as "find out if WordPress is currently displaying any snackbar-style notices".

haszari commented on client/merchant-feedback-prompt/index.tsx:
```diff
@@ -0,0 +1,226 @@
+	const hasRecordedViewEvent = useRef( false );
+
+	// Only render the prompt if there are no core notices.
+	const shouldShowPrompt = coreNotices?.length === 0;
```
If the coast is clear, we can show our (hacky) notice; otherwise we stay out of the way of WordPress notices.

haszari commented on client/merchant-feedback-prompt/index.tsx:
```diff
@@ -0,0 +1,226 @@
+	useEffect( () => {
+		// Record the event when the prompt is rendered, but only once per screen.
+		if ( shouldShowPrompt && ! hasRecordedViewEvent.current ) {
+			recordEvent( 'wcpay_merchant_feedback_prompt_view' );
```
Should we include the screen as a prop? Ideally we'd use `screen_id`, but we could use path or hard-code a screen "name".

If we track this and some screens work better than others, we can optimise how we interrupt merchants.

haszari replied:
> Record the event when the prompt is rendered, but only once per screen.

Why once per screen? This seems unnecessarily complicated. Tracks events normally fire every time the thing happens, why is this one "smart" in this way?

haszari replied:
Hmm … maybe it's more to ensure we only fire one event for each render? If so I approve (but the comment is confusing!).

brucealdridge replied:
I assumed that recording an event saved the page or path. 
The full url is stored under prop `browserdocumentlocation` eg `https://woo.test/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview`
But simplifying this is a `screen` or `page` prop would make it easier to understand performance.

haszari replied:
@nagpai and I were discussing this last week, we need a standard way to track "which screen" events fire on. Generally there is agreement that Woo/WordPress `get_current_screen()` would work well, but it's broken in WooPayments. 

Check out this issue and PR review for more info: 

- https://github.com/Automattic/woocommerce-payments/issues/10439
- https://github.com/Automattic/woocommerce-payments/pull/10411/files#r1967186202

Maybe we can handle this as follow up (in this project, not this PR):

1. Implement / fix `screen_id` in WooPayments.
2. Use it for these tracks events.

Jinksi replied:
> Maybe we can handle this as follow up (in this project, not this PR):

I've added this as a consideration for project issue #10331. Since it is already available as `browserdocumentlocation`, I wouldn't want to delay the release of this project for it.

Jinksi replied:
> Why once per screen? This seems unnecessarily complicated.

This was to prevent the event re-triggering when `shouldShowPrompt` updates (core notices are added/removed). But I agree this is over-engineered and I've removed the ref tracking in 78a56c66fb76c1e9ca81a8182bc59d48d4916715.

Jinksi replied:
I've clarified and simplified these temporary workaround code comments in 8879ba9131b5f809fa65e15ed768757351fa8da0

haszari commented on client/merchant-feedback-prompt/index.tsx:
```diff
@@ -0,0 +1,226 @@
+						 *
+						 * This temporary workaround will remain in place until either:
+						 * - Gutenberg Snackbar component is updated to accept two actions and we can use `core/notices` `createNotice()` to render the snackbar, or
+						 * - The campaign for this prompt is complete and we can remove this code entirely.
```
Might as well document this in one place, if there are two copies then in it's a disincentive to make the explanation clear or tweak / adjust it.

Jinksi replied:
I'll tidy up these comments into a single location to make it more readable and easier to modify.

There are different hacks for different limitations:
* React portal can be removed once `core/notices` `createNotice()` accepts 2 actions, regardless of WooPayments unbundling.
* Custom snackbar component can be removed once `@wordpress/components` Snackbar accepts 2 actions _**and**_ WooPayments unbundles/updates to the updated library version.
* `explicitDismiss` can be removed once WooPayments unbundles or updates to the **_current_** `@wordpress/components` version

haszari commented on client/merchant-feedback-prompt/index.tsx:
```diff
@@ -0,0 +1,226 @@
+ * This is used to ensure the prompt is only rendered if the account is eligible for the campaign and the user has not dismissed the prompt.
+ */
+export default function MaybeShowMerchantFeedbackPrompt() {
+	// TODO: This is a temporary local state to track if the prompt has been dismissed. Move to a user-persistent state in #10329.
```
Grammar nitpick – don't need `a` for "state" – this is temporary local state, move to user-persistent state. "State" is a continuous quantity.

Jinksi replied:
Thanks for picking this up. This comment will be removed immediately in the subsequent PR #10444.

haszari commented on client/merchant-feedback-prompt/index.tsx:
```diff
@@ -0,0 +1,226 @@
+	// TODO: This is a temporary local state to track if the prompt has been dismissed. Move to a user-persistent state in #10329.
+	const [ hasUserDismissed, setHasUserDismissed ] = useState( false );
+
+	const isAccountEligible =
```
This is probably clearer as early-return logic.

```js
if ( the store is a prod store and not opted in to testing our code before we're ready to release ) { return and render nothing }:

if ( the store is not within the campaign criteria ) { return and render nothing }:

if ( the merchant has dismissed or responded ) { return and render nothing }:

// (else)

// ask for feedback
```

Jinksi replied:
This logic is shuffling around in #10444 where I've opted for a hook to encapsulate the visibility logic, so I'll look into making suitable readability changes there.

Commented: https://github.com/Automattic/woocommerce-payments/pull/10444/files#r1976912524

haszari commented on client/merchant-feedback-prompt/style.scss:
```diff
@@ -0,0 +1,25 @@
+// This is a temporary style to ensure the prompt is rendered in the WC footer.
+// It should be removed once Gutenberg SnackbarList component accepts two actions and WooPayments can use the new component.
```
Similar to other comments, I think it's clearer to lean into "this code is temporary" since the update to use core stuff will probably be after we're done. And recommend having this info / comment in one place so message is not diluted.

Jinksi replied:
Updated in f652c05

haszari commented on client/merchant-feedback-prompt/style.scss:
```diff
@@ -0,0 +1,25 @@
+
+.wcpay-merchant-feedback-prompt-wrap {
+	// Borrowed from https://github.com/woocommerce/woocommerce/blob/7e7916106abc835c16388ac24097ba390a5f8d41/plugins/woocommerce/client/admin/client/layout/transient-notices/style.scss
+	position: absolute;
```
Nice to see the links to where the code was borrowed from 👌

haszari commented on includes/class-wc-payments-features.php:
```diff
@@ -393,4 +395,15 @@ private static function is_woopayments_gateway_enabled() {
+	/**
+	 * Checks if the merchant feedback prompt development feature flag is enabled.
+	 *
+	 * This is a temporary development flag that will be removed once the feature is ready for production.
```
Minor thought, I'd flip this comment to explain what the flag is for, since the `checks if` part is a standard convention, very mechanical.

Jinksi replied:
Updated in a42d91b2109bdbb458d1f7cf43c9a36318d5e7a2

haszari reviewed (approved):
Looks good! I see this has been already tested by @jessy-p  and has good feedback already addressed.

No blockers from me, just some more thoughts on clarity of comments & logic flow. Let's get this merged and start polishing the whole flow! If for some reason there are issues here we are protected by the dev flag anyway.

haszari commented on client/merchant-feedback-prompt/index.tsx:
```diff
@@ -0,0 +1,226 @@
+ *
+ * This is used to ensure the prompt is only rendered if the account is eligible for the campaign and the user has not dismissed the prompt.
+ */
+export default function MaybeShowMerchantFeedbackPrompt() {
```
Naming nitpick – when this is imported / used it's referred to as `MerchantFeedbackPrompt`, which is confusing, since the actual prompt with no conditionals is `MerchantFeedbackPrompt`.

Not a blocker, but might be clearer to:

- Use `Maybe` naming in all client page reference, so it's obvious that `<MaybeShowMerchantFeedbackPrompt /> will manage the visibility.
- OR
- Renamed the inner component to something other than `MerchantFeedbackPrompt`.

Jinksi replied:
Great idea. I've changed to use `Maybe` naming (and changed to named exports) when used to clarify this behaviour in 67c6ec917a47ec59ae3bbe44e69c03251dfe3471.

https://github.com/Automattic/woocommerce-payments/blob/67c6ec917a47ec59ae3bbe44e69c03251dfe3471/docs/typescript/general-conventions.md#L3-L17

brucealdridge commented on client/disputes/index.tsx:
```diff
@@ -446,6 +447,7 @@ export const DisputesList = (): JSX.Element => {
 
 	return (
 		<Page>
+			<MerchantFeedbackPrompt />
```
If we're being selective about which pages we put this prompt on, I wonder if we should avoid disputes.

haszari replied:
🤔

Jinksi replied:
I'll leave this decision to another project issue #10481

brucealdridge reviewed (approved):
The prompt animates in on every page navigation that seems out of place. 
Is this intentional? It is attention grabbing but I think its more likely to be dimissed with this behavior.

https://github.com/user-attachments/assets/a42090fe-4fcb-474e-b301-835dc18e0817

Tracks events are being recorded correctly ✅

Jinksi commented:
>  It is attention grabbing but I think its more likely to be dimissed with this behavior.
> @brucealdridge 

Not necessarily intentional – this is a side-effect of not using the wp-admin-native `createNotice()` and the way I've implemented it, resulting in a re-render each page navigation. I'll see if I can tweak this PR to remove this aggressive re-animation.

Jinksi committed:
Change component name to MaybeShowMerchantFeedbackPrompt to clarify behaviour

Jinksi committed:
Simplify merchant feedback prompt view event tracking

Remove unnecessary ref tracking for view event recording, ensuring the event is tracked directly when the prompt should be shown

Jinksi committed:
Remove unused `useRef` import

haszari commented:
> Not necessarily intentional – this is a side-effect of not using the wp-admin-native createNotice() and the way I've implemented it, resulting in a re-render each page navigation. I'll see if I can tweak this PR to remove this aggressive re-animation.

@Jinksi that could be handled as a follow up issue if you prefer. A dedicated issue would allow us to brainstorm / discuss what behaviour we want – design that part of how the prompt works.

Jinksi committed:
Simplify and clarify temporary workaround code comments

Jinksi committed:
Clarify feature flag code comment

Jinksi committed:
Update comment for merchant feedback prompt temporary style

Jinksi committed:
Link to issue in dev feature flag code comment

Jinksi commented:
> A dedicated issue would allow us to brainstorm / discuss what behaviour we want – design that part of how the prompt works.

I've opened #10482 for this
